libraries/ESP8266mDNS/examples/LEAmDNS/mDNS_ServiceMonitor/mDNS_ServiceMonitor.ino

```diff
@@ -212,18 +249,20 @@ bool MDNSProbeResultCallback(MDNSResponder* p_pMDNSResponder,
 
         if (!hMDNSService) {
           // Add a 'http.tcp' service to port 'SERVICE_PORT', using the host domain as instance domain
-          hMDNSService = p_pMDNSResponder->addService(0, "http", "tcp", SERVICE_PORT);
+          hMDNSService = MDNS.addService(0, "http", "tcp", SERVICE_PORT);
```

Comment:
In general I agree with you on using global instances.
However in this case it is not a user defined global but the one which comes with the MDNS implementation.  So other calls in the app also use global MDNS
If a user does want the MDNS* in the callback he/she can  use the functional to get that in the callback.

```
bool hostProbeResult (MDNSResponder* callMDNS, String p_pcDomainName, bool p_bProbeResult)
{
  callMDNS->addService()
}

MDNS.setHostProbeResultCallback(std::bind(hostProbeResult(&MDNS, _1, _2));
or 
MDNSResponder myDMNS;
myMDNS.setHostProbeResultCallback(std::bind(hostProbeResult(&myMDNS, _1, _2));
```

My intention was to keep the callback signature as simple as possible.

Reply:
Users tend to be guided by our examples way too often. That means we should stick to good practices, so that users adopt them. How about changing the example to what you just suggested?

Reply:
Finalizing should start now. I can do either way.

Current : 
```
    typedef std::function<bool (String p_pcDomainName,
                                bool p_bProbeResult)> MDNSHostProbeResultCallbackFn;
```
Usage : 
```
bool hostProbeResult (String p_pcDomainName, bool p_bProbeResult)
{
  MDNS.addService()
}
MDNS.setHostProbeResultCallback(hostProbeResult);
```
or
```
bool hostProbeResult (MDNSResponder* callMDNS, String p_pcDomainName, bool p_bProbeResult)
{
  callMDNS->addService()
}
MDNS.setHostProbeResultCallback(std::bind(hostProbeResult(&MDNS, _1, _2));
```
Alternative 
```
    typedef std::function<bool (MDNSResponder* callMDNS, 
                                String p_pcDomainName,
                                bool p_bProbeResult)> MDNSHostProbeResultCallbackFn;
```
Usage : 
```
bool hostProbeResult (String p_pcDomainName, bool p_bProbeResult)
{
  MDNS.addService()
}
MDNS.setHostProbeResultCallback(std::bind(hostProbeResult(_2,_3);
```
or : 
```
bool hostProbeResult (MDNSResponder* callMDNS, String p_pcDomainName, bool p_bProbeResult)
{
  callMDNS->addService()
}
MDNS.setHostProbeResultCallback(hostProbeResult);
```
I have chosen for the current as I expect that people wanting to have a the callDNS are more experienced and less trouble getting into the std::bind() functionality.

In any case we should have both options in an example.